### PR TITLE
Add icons, links, and tests to directory table

### PR DESCRIPTION
**Pull request by mattgraham1995**
image

Created a new component for the name column cells in the table. This component handles the item's icon, the title, the link, and the favorite icon. Later on, I will add the actions list to this column as well.

### CODE REVIEW

File: packages/directory/tests/dummy/mirage/bard-lite/dimensions/displayCurrency.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+export default [
```
[the remark this answers is not in the document]
Comment (mattgraham1995) [reply packages/directory/tests/dummy/mirage/bard-lite/dimensions/displayCurrency.js]: Just so the test reports in the fixtures actually get some results.

File: packages/directory/addon/components/dir-table.js [hunk 2]
```diff
@@ -53,14 +53,10 @@ export default Component.extend({
...
-      let type = get(item.serialize().data, 'type');
 
       return {
-        type,
-        id: get(item, 'id'),
-        name: get(item, 'title'),
-        lastUpdatedDate: Moment(get(item, 'updatedOn')).format('MM/DD/YYYY -  hh:mm:ss a z'),
-        author: get(item, 'author.id')
+        model: item,
+        lastUpdatedDate: Moment(get(item, 'updatedOn')).format('MM/DD/YYYY -  hh:mm:ss a')
```
[the remark this answers is not in the document]
Comment (mattgraham1995) [reply packages/directory/addon/components/dir-table.js]: Yes, but the format wasn't correct.

[the remark this answers is not in the document]
Comment (balajimanoharan) [reply packages/directory/addon/components/dir-table.js]: Can we do this with the `moment-format` helper?

File: packages/directory/addon/components/dir-item-name-cell.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+   * @property {String} type - the type of the item
+   */
+  type: computed('value', function() {
+    return get(this, 'value').serialize().data.type;
+  }),
+
+  /**
+   * @property {String} iconClass - the icon class that is passed to navi-icon
+   */
+  iconClass: computed('value', function() {
```
Comment (Xuefeng-Zhu) [inline packages/directory/addon/components/dir-item-name-cell.js]: typo here?

File: packages/directory/app/styles/navi-directory/components/dir-item-name-cell.less [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Copyright 2018, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+ .dir-item-name-cell {
```
Comment (Xuefeng-Zhu) [inline packages/directory/app/styles/navi-directory/components/dir-item-name-cell.less]: extra space here

File: packages/directory/addon/components/dir-table.js [hunk 5]
```diff
@@ -71,21 +67,33 @@ export default Component.extend({
...
+      breakpoints: ['desktop', 'jumbo']
     }];
   }),
 
   /**
    * @property {Object} table - Used by ember-light-table to create the table
    */
   table: computed('model', function() {
-    return new Table(this.get('columns'), this.get('model'));
+    return new Table(this.get('columns'), this.get('model'), {
```
Comment (Xuefeng-Zhu) [inline packages/directory/addon/components/dir-table.js]: Do we need to add `columns` to observed list?

**Pull request merged**